### PR TITLE
Upgrade to DCL Version 1.37.0

### PR DESCRIPTION
**Pull request by hao-nan-li**
Upgrade to DCL Version 1.37.0

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
clouddeploy: added fields for resource `google_clouddeploy_delivery_pipeline`
```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
Looks like this dcl release will introduce some untested fields, and some unexpected errors. The errors could be fixed locally at the downstream, but it will appear again on another make teraaform.

File: tpgtools/api/clouddeploy/samples/updatecanary.delivery_pipeline.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+    "name": "{{resource_name}}",
```
Comment (c2thorn) [inline tpgtools/api/clouddeploy/samples/updatecanary.delivery_pipeline.json:2]:
```suggestion
    "name": "{{pipeline}}",
```
Should fix the current VCR error

Comment (hao-nan-li) [reply tpgtools/api/clouddeploy/samples/updatecanary.delivery_pipeline.json:2]: Thanks for the suggestion, I'll run the test couple times locally to make sure that's all the remaining problem.

**Comment by trodge** [comment]
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7619-old..auto-pr-7619) ( 8 files changed, 406 insertions(+), 9 deletions(-)) Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7619-old..auto-pr-7619) ( 8 files changed, 1158 insertions(+), 7 deletions(-)) TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7619-old..auto-pr-7619) ( 2 files changed, 6 insertions(+), 6 deletions(-)) TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7619-old..auto-pr-7619) ( 2 files changed, 6 insertions(+), 6 deletions(-))
> 
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_clouddeploy_delivery_pipeline` (10 total tests) Untested fields: `serial_pipeline.stages.strategy.canary.custom_canary_deployment.phase_configs.percentage`, `serial_pipeline.stages.strategy.canary.custom_canary_deployment.phase_configs.phase_id`, `serial_pipeline.stages.strategy.canary.custom_canary_deployment.phase_configs.profiles`, `serial_pipeline.stages.strategy.canary.custom_canary_deployment.phase_configs.verify`
> 
> Please add acceptance tests which include these fields.

The missing test detector is missing fields covered by nested objects in arrays.

https://github.com/GoogleCloudPlatform/magic-modules/pull/7614 will fix this issue in future.

For now these findings can be ignored.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**